src/Roslyn.Suppressions/Roslyn.Suppressions/RoslynSettingsFileInfo.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions
+{
+    internal static class RoslynSettingsFileInfo
+    {
+        public static readonly string Directory = Path.Combine(Path.GetTempPath(), "SLVS", "Roslyn");
+
+        public static string GetFilePathFromEscapedProjectKey(string sonarProjectKey)
```

Comment:
maybe name the variable as escaped project key so it would be obvious for later.

Reply:
Why not move the escaping into this class too? That way the caller(s) don't have to know how to escape it.

Reply:
We can't reverse it back so one method would be `GetFilePathFromProjectKey` and the other would be `GetEscapedProjectKeyFromFilePath`, which is a bit weird.

Reply:
It is a bit odd, but it doesn't seem right that the caller needs to know how to do part of the name-mangling calculation.

Yep, perhaps just changing the naming would be enough e.g. `projectMoniker`, `projectToken`, `projectId`, `settingsKey`

```
// Returns the full file path for the given project key
string GetSettingsFilePath(string sonarProjectKey)


// Returns an identifier for the project settings to which the settings file relates.
// <remarks>The identifier is *not* the actual project key since we can't recover it accurately from the file name </remarks> 
string GetSettingsKey(string settingsFilePath)

```

Reply:
Nice - the updated version looks good.